nautobot/core/models/tree_queries.py

```diff
@@ -28,6 +30,10 @@ class TreeManager(TreeManager_, BaseManager.from_queryset(TreeQuerySet)):
     _with_tree_fields = True
     use_in_migrations = True
 
+    @cached_property
```

Comment:
There's a chance that old data will be retrieved if a new location is added, which could change `max_tree_depth`.

Is it possible whenever we create a location with a descendent length larger than the current `max_depth`, we should invalidate the cache.